pallets/xcm-transactor/src/lib.rs

```diff
@@ -848,6 +876,198 @@ pub mod pallet {
 		}
 	}
 
+	impl<T: Config> UtilityEncodeCall for Pallet<T> {
+		fn encode_call(self, call: UtilityAvailableCalls) -> Vec<u8> {
+			match call {
+				UtilityAvailableCalls::AsDerivative(a, b) => {
+					let mut encoded_call: Vec<u8> = Vec::new();
+					// pallet index
+					encoded_call.push(RelayIndices::<T>::get().pallets.utility);
+					// call index
+					encoded_call.push(RelayIndices::<T>::get().calls.utility.as_derivative);
+					// encoded argument
+					encoded_call.append(&mut a.encode());
+					encoded_call.append(&mut b.clone());
+					encoded_call
+				}
+			}
+		}
+	}
+
+	impl<T: Config> HrmpEncodeCall for Pallet<T> {
+		fn hrmp_encode_call(call: HrmpAvailableCalls) -> Result<Vec<u8>, xcm::latest::Error> {
+			match call {
+				HrmpAvailableCalls::InitOpenChannel(a, b, c) => {
+					let mut encoded_call: Vec<u8> = Vec::new();
+					// pallet index
+					encoded_call.push(RelayIndices::<T>::get().pallets.hrmp);
+					// call index
+					encoded_call.push(RelayIndices::<T>::get().calls.hrmp.init_open_channel);
+					// encoded arguments
+					encoded_call.append(&mut a.encode());
+					encoded_call.append(&mut b.encode());
+					encoded_call.append(&mut c.encode());
+					Ok(encoded_call)
+				}
+				HrmpAvailableCalls::AcceptOpenChannel(a) => {
+					let mut encoded_call: Vec<u8> = Vec::new();
+					// pallet index
+					encoded_call.push(RelayIndices::<T>::get().pallets.hrmp);
+					// call index
+					encoded_call.push(RelayIndices::<T>::get().calls.hrmp.accept_open_channel);
+					// encoded argument
+					encoded_call.append(&mut a.encode());
+					Ok(encoded_call)
+				}
+				HrmpAvailableCalls::CloseChannel(a) => {
+					let mut encoded_call: Vec<u8> = Vec::new();
+					// pallet index
+					encoded_call.push(RelayIndices::<T>::get().pallets.hrmp);
+					// call index
+					encoded_call.push(RelayIndices::<T>::get().calls.hrmp.close_channel);
+					// encoded argument
+					encoded_call.append(&mut a.encode());
+					Ok(encoded_call)
+				}
+				HrmpAvailableCalls::CancelOpenRequest(a, b) => {
+					let mut encoded_call: Vec<u8> = Vec::new();
+					// pallet index
+					encoded_call.push(RelayIndices::<T>::get().pallets.hrmp);
+					// call index
+					encoded_call.push(RelayIndices::<T>::get().calls.hrmp.cancel_open_request);
+					// encoded argument
+					encoded_call.append(&mut a.encode());
+					encoded_call.append(&mut b.encode());
+					Ok(encoded_call)
+				}
+			}
+		}
+	}
+
+	fn encode_compact_arg<T: parity_scale_codec::HasCompact>(input: T) -> Vec<u8> {
+		#[derive(Encode)]
+		struct CompactWrapper<T: parity_scale_codec::HasCompact> {
+			#[codec(compact)]
+			input: T,
+		}
+		CompactWrapper { input }.encode()
+	}
+
+	impl<T: Config> StakeEncodeCall for Pallet<T> {
```

Comment:
I would move this Impls to its own file, otherwise xcm-transactor will become unreadable

Reply:
makes sense, moved to the new `encode.rs`